stdlib/3/tempfile.pyi

```diff
@@ -93,17 +92,48 @@ def NamedTemporaryFile(
 # It does not actually derive from IO[AnyStr], but it does implement the
 # protocol.
 class SpooledTemporaryFile(IO[AnyStr]):
-    def __init__(self, max_size: int = ..., mode: str = ...,
-                 buffering: int = ..., encoding: Optional[str] = ...,
-                 newline: Optional[str] = ..., suffix: Optional[str] = ...,
-                 prefix: Optional[str] = ..., dir: Optional[str] = ...
-                 ) -> None: ...
+    # bytes needs to go first, as default mode is to open as bytes
+    @overload
+    def __init__(
+        self: SpooledTemporaryFile[bytes],
+        max_size: int = ...,
+        mode: Literal["rb", "wb", "ab", "xb", "r+b", "w+b", "a+b", "x+b"] = ...,
```

Comment:
Also just `"b"`.

Reply:
The mode `"b"` and `""` both cause errors in Python 2.7 and 3.7. I haven't tested on other Python versions, but I don't see why they would be different.

Trying to instantiate a `TemporaryFile` (done when a `SpooledTemporaryFile` rolls over) with a mode string not containing exactly one of `rwax` and more than one `+` will raise `ValueError: Must have exactly one of create/read/write/append mode and at most one plus` on Python 3.7.4.

Python 2.7.16 has a similar error: `ValueError: mode string must begin with one of 'r', 'w', 'a' or 'U', not 'b'`.

Confusingly, the `SpooledTemporaryFile` constructor accepts any mode, but an exception is raised only when it creates a `TemporaryFile` internally. [This can be forced by using the `rollover()` function](https://docs.python.org/3/library/tempfile.html#tempfile.SpooledTemporaryFile).

Test case:

```python3
import tempfile
# this line works fine
with tempfile.SpooledTemporaryFile(mode="b") as tmpfile:
    # this line fails, since the invalid mode is checked here
    tmpfile.rollover()
```